vscode-lean4/test/suite/lakefileTomlSchema/lakefileTomlSchema.test.ts

```diff
@@ -0,0 +1,76 @@
+import assert from 'assert'
+import * as fs from 'fs'
+import { suite } from 'mocha'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { logger } from '../../../src/utils/logger'
+import { sleep, waitForActiveExtension } from '../utils/helpers'
+
+suite('Tests', () => {
+    const extensionDevelopmentPath = path.resolve(__dirname, '..', '..', '..', '..')
+    const testCaseLocation = path.join(extensionDevelopmentPath, 'test', 'test-fixtures', 'lakefileTomlSchemaTestCases')
+    const validLakefilesDirectory = path.join(testCaseLocation, 'valid')
+    const invalidLakefilesDirectory = path.join(testCaseLocation, 'invalid')
+
+    for (const testFileName of fs.readdirSync(invalidLakefilesDirectory)) {
+        const testFileLocation = path.join(invalidLakefilesDirectory, testFileName)
+        test(testFileName, async () => {
+            logger.log(`=================== Ensure ${testFileName} is rejected ===================`)
+            assert(vscode.workspace.workspaceFolders !== undefined, 'No workspace folder is opened')
+            assert(vscode.workspace.workspaceFolders.length === 1, 'Exactly one workspace folder should be opened ')
+            const workspaceFolder = vscode.workspace.workspaceFolders[0].uri.fsPath
+            logger.log(`Found workspace folder at ${workspaceFolder}`)
+            const lakefilePath = path.join(workspaceFolder, 'lakefile.toml')
+
+            assert(fs.existsSync(testFileLocation), `Test case location does not exist: ${testFileLocation}`)
+
+            fs.copyFileSync(testFileLocation, lakefilePath)
+
+            const document = await vscode.workspace.openTextDocument(lakefilePath)
+            await vscode.window.showTextDocument(document)
+
+            await waitForActiveExtension('tamasfe.even-better-toml')
+
+            // Wait for 5 seconds for diagnostics to appear
+            await sleep(5 * 1000)
+
+            const diagnostics = vscode.languages.getDiagnostics().flatMap(([, diagnostic]) => diagnostic)
```

Comment:
I wonder if there is something a little more robust we could do here, like listening for diagnostic changes before opening the document and then using the first set of diagnostics that is reported by even-better-toml (or timing out after 60s), which would rely on the assumption that even-better-toml only reports diagnostics once, but the limitations of the VS Code API ensure that we won't really have a perfectly robust solution here either way.

If the 5s delay works consistently, then I'm fine with it :-)